Thông tin các bộ dữ liệu:

- `car-sales.csv`: bộ dữ liệu chứa thông tin của các ôtô cũ ở một cửa hàng


### 1. Đọc dữ liệu

# Lab 02 - Bài tập

---

## THÔNG TIN

- Họ tên: Trần Gia Vỹ
- MSV: 23715831
- Ngày hoàn thành:11/01/2025

---
## NUMPY

Bạn hãy phát sinh ngẫu nhiên mảng ndarray có tên `arr` gồm 100 phần tử số nguyên có giá trị từ 0..1000. Thực hiện các thao tác sau:

### 1. Hiển thị các phần tử thỏa yêu cầu dưới đây

- 10 phần tử đầu tiên
- 5 phần tử cuối cùng
- Các phần tử thỏa mãn điều kiện chia hết cho 5 và lớn hơn 100

In [ ]:
import numpy as np
arr = np.random.randint(0,1001, size=100)
print("Muoi phan tu dau tien:",arr[:10])
print("Nam phan tu cuoi cung:",arr[-5:])
a = arr[( arr%5 == 0) & (arr>100)]
print("Cac phan tu thoai man chia het cho 5 va lon hon 100:",a)

Muoi phan tu dau tien: [409 874 760  16 190 142 922 851  77 280]
Nam phan tu cuoi cung: [164 977 874 777 192]
Cac phan tu thoai man chia het cho 5 va lon hon 100: [760 190 280 500 720 360 245 245 750 470 935 640 650 775 190 520]


### 2. Cho biết mảng trên có bao nhiêu phần tử có giá trị nhỏ hơn 100. Hiển thị các phần tử đó

In [ ]:
import numpy as np
arr = np.random.randint(0,1001, size=100)
a = arr[(arr<100)]
print("Cac phan tu nho hon 100",a)

Cac phan tu nho hon 100 [19 27  3 81 83 49 35 20 38]


### 3. Thực hiện thống kê:

- Tính tổng mảng
- Tìm giá trị trung bình
- Tìm giá trị trung vị
- Tìm Q1, Q2, Q3
- Tìm phân vị 20 (percentile 20)

In [ ]:
import numpy as np
import math
arr = np.random.randint(0,1001, size=100)
tong = arr.sum()
print("Tong mang:",tong)
trung_binh = arr.mean()
print("Giá trị trung bình:",trung_binh)
Trung_vi = np.median(arr)
print("Giá trị trung vị:", Trung_vi)
print('-Phân vị Q1 (median): ', np.quantile(arr,0.25))
print('-Phân vị Q2 (median): ', np.quantile(arr,0.5))
print('-Phân vị Q3 (median): ', np.quantile(arr,0.75))
Phan_vi_20 = np.percentile(arr, 20)
print("-Phân vị 20:", Phan_vi_20)

Tong mang: 47574
Giá trị trung bình: 475.74
Giá trị trung vị: 448.5
-Phân vị Q1 (median):  258.75
-Phân vị Q2 (median):  448.5
-Phân vị Q3 (median):  728.5
Phân vị 20: 209.0


---

## PANDAS

Thông tin các bộ dữ liệu:

- `car-sales.csv`: bộ dữ liệu chứa thông tin của các ôtô cũ ở một cửa hàng


### 1. Đọc dữ liệu

In [2]:
import pandas as pd
Doc = pd.read_csv('car-sales.csv',header = 0)
Doc.head()

Manufacturer Colour  Odometer  Doors    Price
0       Toyota  White    150043      4   4000.0
1        Honda    Red     87899      4   5000.0
2       Toyota   Blue     32549      3   7000.0
3          BMW  Black     11179      5  22000.0
4       Nissan  White    213095      4   3500.0

### 2. Dữ liệu có bao nhiêu dòng, bao nhiêu cột

Đáp án: (10 dòng, 5 cột)

In [4]:
dong,cot = Doc.shape
print (f" Du lieu co {dong} dong va co {cot} cot")

 Du lieu co 10 dong va co 5 cot


- Định tính (phân loại): Manufacturer, Colour
- Định lượng: Odemeter, Price, Doors

In [7]:
DT = ['Manufacturer','Colour']
DL = ['Odometer','Doors','Price']
print(Doc[DT])
print(Doc[DL])

  Manufacturer Colour
0       Toyota  White
1        Honda    Red
2       Toyota   Blue
3          BMW  Black
4       Nissan  White
5       Toyota  Green
6        Honda   Blue
7        Honda   Blue
8       Toyota  White
9       Nissan  White
   Odometer  Doors    Price
0    150043      4   4000.0
1     87899      4   5000.0
2     32549      3   7000.0
3     11179      5  22000.0
4    213095      4   3500.0
5     99213      4   4500.0
6     45698      4   7500.0
7     54738      4   7000.0
8     60000      4   6250.0
9     31600      4   9700.0


### 3. Trong dữ liệu có thông tin của bao nhiêu xe?

Đáp án: 10 xe

In [10]:
so_xe = Doc.shape[0]
print(f"trong du lieu co so xe {so_xe}")

trong du lieu co so xe 10


### 4. Đếm số xe theo theo hãng

Đáp án: Toyota: 10 xe, Honda: 3 xe, Nissan: 2 xe, BMW: 1 xe

In [13]:
Dem = Doc['Manufacturer'].value_counts()
print(Dem)

Manufacturer
Toyota    4
Honda     3
Nissan    2
BMW       1
Name: count, dtype: int64


### 5. Đếm số xe theo màu


In [16]:
Dem = Doc['Colour'].value_counts()
print(Dem)

Colour
White    4
Blue     3
Red      1
Black    1
Green    1
Name: count, dtype: int64


### 6. Tổng giá trị các xe trong cửa hàng

Đáp án: 76450 USD


In [19]:
Tong = Doc['Price'].sum()
print(Tong)

76450.0


### 7. Top 3 xe có giá cao nhất


In [22]:
Top_3 = Doc.nlargest(3, 'Price')
print(Top_3)

  Manufacturer Colour  Odometer  Doors    Price
3          BMW  Black     11179      5  22000.0
9       Nissan  White     31600      4   9700.0
6        Honda   Blue     45698      4   7500.0


### 8. Top 3 xe đã chạy nhiều cây số (km)nhất (Odometer)

Đáp án:
- Xe Nissan, White, 213095 km
- Xe Toyota, White, 150043 km
- Xe Toyota, Green, 99213 km

In [25]:
Top_3 = Doc.nlargest(3, 'Odometer')
print(Top_3)

  Manufacturer Colour  Odometer  Doors   Price
4       Nissan  White    213095      4  3500.0
0       Toyota  White    150043      4  4000.0
5       Toyota  Green     99213      4  4500.0


### 9. Khảo sát xe của hãng Honda

- Tính tỷ lệ xe Honda trong dữ liệu của cửa hàng (0.3)
- Cho biết giá trung bình của xe hãng Honda của cửa hàng (6500)
- Cho biết các xe của hãng Honda trong cửa hàng có những màu sắc nào (Red, Blue)

In [28]:
Honda = Doc[Doc["Manufacturer"]== 'Honda']
TL = len(Honda)/len(Doc)
print(f"Ti le xe Honda :{TL}")
GiaTB = Honda['Price'].mean()
print(f"Gia trung binh cua xe Hoanda:{GiaTB}")
Mau = Honda['Colour'].value_counts()
print(f"Mau cua cac loai xe:{Mau}")

Ti le xe Honda :0.3
Gia trung binh cua xe Hoanda:6500.0
Mau cua cac loai xe:Colour
Blue    2
Red     1
Name: count, dtype: int64


### 10. Tính toán các giá trị sau của  thuộc tính giá ('Price')

- Giá trung bình các xe (7645)
- Giá trị trung vị các xe (6625)
- Miền giá trị của xe (18500)
- Miền phân vị của xe (2750)
- Điểm phân vị 0.4 và 0.6 của giá ([5750, 7000])

In [31]:
GiaTB = Doc['Price'].mean()
print(f"Gia trung binh cua xe Hoanda:{GiaTB}")
GiaTV = Doc['Price'].median()
print(f"Gia trung vi cua xe Hoanda:{GiaTV}")
Min = Doc['Price'].min()
Max = Doc['Price'].max()
Mien_GT = Max - Min
print(f"Mien gia tri cua xe {Mien_GT}")
q1 = Doc['Price'].quantile(0.25)
q3 = Doc['Price'].quantile(0.75)
Mien_PV = q3 - q1
print(f"Mien phan vi cua xe {Mien_PV}")
q04 = Doc['Price'].quantile(0.4)
q06 = Doc['Price'].quantile(0.6)
print(f"Diem phan vi 0.4 co gia {q04} va diem phan vi 0.6 cos gia {q06}")

Gia trung binh cua xe Hoanda:7645.0
Gia trung vi cua xe Hoanda:6625.0
Mien gia tri cua xe 18500.0
Mien phan vi cua xe 2750.0
Diem phan vi 0.4 co gia 5750.0 va diem phan vi 0.6 cos gia 7000.0


---